pkg/compiler/compiler.go

```diff
@@ -0,0 +1,119 @@
+package compiler
+
+import (
+	"errors"
+	"fmt"
+
+	"evylang.dev/evy/pkg/code"
+	"evylang.dev/evy/pkg/evaluator"
+	"evylang.dev/evy/pkg/parser"
+)
+
+// ErrUndefinedVar is returned when a variable name cannot
+// be resolved in the symbol table.
+var ErrUndefinedVar = errors.New("undefined variable")
+
+// Compiler is responsible for turning a parsed evy program into
+// bytecode.
+type Compiler struct {
+	constants    []evaluator.Value
+	instructions code.Instructions
+	symbolTable  *SymbolTable
+}
+
+// Bytecode represents raw evy bytecode.
+type Bytecode struct {
+	Constants    []evaluator.Value
+	Instructions code.Instructions
+}
+
+// New returns a new compiler.
+func New() *Compiler {
+	return &Compiler{
+		constants:    []evaluator.Value{},
+		instructions: code.Instructions{},
+		symbolTable:  NewSymbolTable(),
+	}
+}
+
+// Compile accepts an AST node and renders it to bytecode internally.
+func (c *Compiler) Compile(node parser.Node) error {
+	switch node := node.(type) {
+	case *parser.Program:
+		return c.compileProgram(node)
+	case *parser.InferredDeclStmt:
+		return c.compileDecl(node.Decl)
+	case *parser.AssignmentStmt:
+		return c.compileAssignment(node)
+	case *parser.Var:
+		return c.compileVar(node)
+	case *parser.NumLiteral:
+		integer := &evaluator.NumVal{V: node.Value}
```

Comment:
`integer` is a bad name, rename to `num` and add a test to prove that we support floating point numbers

Reply:
Agreed.